Mathlib/Tactic/UnitInterval.lean

```diff
@@ -0,0 +1,6 @@
+/-- A tactic that solves `0 ≤ ↑x`, `0 ≤ 1 - ↑x`, `↑x ≤ 1`, and `1 - ↑x ≤ 1` for `x : I`. -/
+macro "unit_interval" : tactic => `(tactic| first
+  | apply unitInterval.nonneg
```

Comment:
```suggestion
  | apply UnitInterval.nonneg
```

The big problem here is that this won't work because there is no `UnitInterval.nonneg` declaration yet.  Due to hygiene, Lean will try to apply a hypothesis called `UnitInterval.nonneg` here (I believe!).  Maybe `set_option hygiene false` helps, but you should try it out with a dummy definition of the unit interval.